llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -854,6 +859,11 @@ PreservedAnalyses SpecConstantsPass::run(Module &M,
 
       Constant *DefaultValue = getSpecConstInitializerFromCI(CI, NameArgNo + 1);
 
+      if (IsSYCLAlloca) {
```

Comment:
I would suggest to move it closer to `SCTy` initialization, so it is easier to spot the difference with the default code path. And you have an assertion in place to make sure that `SCTy` is not overwritten later